### PR TITLE
made doctests compatible and did minor changes to look code more pythonic

### PR DESCRIPTION
**Pull request by shibasisp**

### CODE REVIEW

File: casacore/util/substitute.py [hunk 1]
```diff
@@ -24,17 +24,18 @@
...
+__all__ = ['getlocals', 'getvariable', 'substitute']
+
 
 def getlocals(back=2):
     """Get the local variables some levels back (-1 is top)."""
     import inspect
     fr = inspect.currentframe()
     try:
-        while fr and back != 0:
+        while fr and back:
```
Comment (tammojan) [inline casacore/util/substitute.py]: I think this change makes the code less clear.

Comment (shibasisp) [inline casacore/util/substitute.py]: I will reverse it back.  Rest all are okay?

Comment (tammojan) [inline casacore/util/substitute.py]: Looks good to me!

**Comment by tammojan** [comment]
I updated coveralls to allow a decrease in coverage of 1% or so to avoid silly rejections like this.

**Pull request merged**